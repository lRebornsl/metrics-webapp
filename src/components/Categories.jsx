import { useSelector } from "react-redux";
import forward_icon from "../assets/forward_icon.png";
import categories_icon from "../assets/categories_icons/categories_icon.png";

const Categories = () => {
  const categories = useSelector(state => state.categoriesSlice.categories);

  return(
    <div className="text-white">
      <div className="flex justify-center items-center h-40 bg-united font-bold">
        <img src={categories_icon} alt="Icon" />
        <div>
          <h2 className="text-xl">CATEGORIES</h2>
          <p className="text-end">{categories.length} in total</p>
        </div>
      </div>
      <div className="grid grid-cols-2 bg-queen">
        {categories.map((category, index) => (
          <div className={`p-1 h-50 font-bold ${ !category.bg ? 'bg-queen' : 'bg-yonder' }`} key={index}>
            <div className="flex justify-end items-center">
              <button><img src={forward_icon} alt="Icon" /></button>
            </div>
            <img className="mx-auto py-2" src={category.img} alt="Icon" />
            <h2 className="uppercase text-end">{category.name}</h2>
          </div>
        ))}
      </div>
    </div>
  )
}

export default Categories;