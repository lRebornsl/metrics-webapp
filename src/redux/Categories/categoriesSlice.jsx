import { categoryList } from "../../utils/constant";

const initialState = {
  categories: categoryList,
};

const categoriesSlice = (state = initialState) => state;

export default categoriesSlice;
