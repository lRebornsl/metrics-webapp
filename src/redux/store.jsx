import { configureStore } from '@reduxjs/toolkit';
import categoriesSlice from './Categories/categoriesSlice';

const store = configureStore({
  reducer: {
    categoriesSlice,
  },
});

export default store;