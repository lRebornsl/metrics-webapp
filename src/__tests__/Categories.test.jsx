import { render, waitFor, screen } from "@testing-library/react";
import { beforeEach, describe, expect } from "vitest";
import Categories from "../components/Categories";
import { Provider } from "react-redux";
import store from "../redux/store";
import { MemoryRouter } from "react-router-dom";

describe("Categories tests", () => {
  
  beforeEach(() => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Categories />
        </MemoryRouter>
      </Provider>
    );    
  });

  it("Categories renders correctly", async () => {
    await waitFor(() => expect(Categories).toMatchSnapshot());
  });

  it("Images are rendered correctly", async () => {
    const categoryImages = screen.getAllByRole("img");
    expect(categoryImages).toMatchSnapshot();
  });
})