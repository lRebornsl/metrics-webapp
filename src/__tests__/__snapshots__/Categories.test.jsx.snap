// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Categories tests > Categories renders correctly 1`] = `[Function]`;

exports[`Categories tests > Images are rendered correctly 1`] = `
[
  <img
    alt="Icon"
    src="/src/assets/categories_icons/categories_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/mmorpg_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/shooter_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/strategy_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/moba_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/racing_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/sports_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/social_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/sandbox_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/open_world_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/survival_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/pvp_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/pve_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/pixel_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/voxel_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/zombie_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/turn_based_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/first_person_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/third_person_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/top_down_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/tank_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/space_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/sailing_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/side_scroller_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/superhero_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/permadeath_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/card_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/battle_royale_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/mmo_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/mmo_fps_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/mmo_tps_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/twod_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/threed_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/anime_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/fantasy_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/sci_fi_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/fighting_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/action_rpg_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/action_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/military_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/martial_arts_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/flight_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/low_spec_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/tower_defense_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/horror_icon.png"
  />,
  <img
    alt="Icon"
    src="/src/assets/forward_icon.png"
  />,
  <img
    alt="Icon"
    class="mx-auto py-2"
    src="/src/assets/categories_icons/mmorts_icon.png"
  />,
]
`;
